### PR TITLE
check: support generated cdktf docs

### PR DESCRIPTION
**Pull request by DanielMSchmidt**

### CODE REVIEW

File: check/directory.go [hunk 1]
```diff
@@ -9,28 +9,50 @@ import (
 )
 
 const (
+	CdktfIndexDirectory = `cdktf`
+
 	DocumentationGlobPattern = `{docs/index.md,docs/{data-sources,guides,resources},website/docs}/**/*`
 
 	LegacyIndexDirectory       = `website/docs`
-	LegacyDataSourcesDirectory = `website/docs/d`
```
Comment (bflad) [inline check/directory.go:15]: These constants are exported and therefore part of the Go module compatibility promises, however in spot checking various dependents of this Go module, it does not appear that any drop down to using the Go code directly so it's likely fine that this is being changed.

File: check/directory.go [hunk 2]
```diff
@@ -9,28 +9,50 @@ import (
 )
 
 const (
+	CdktfIndexDirectory = `cdktf`
+
 	DocumentationGlobPattern = `{docs/index.md,docs/{data-sources,guides,resources},website/docs}/**/*`
```
Comment (bflad) [inline check/directory.go:14]: I believe this needs to be updated as well to ensure the checks actually run against `docs/cdktf`:

```suggestion
	DocumentationGlobPattern = `{docs/index.md,docs/{cdktf,data-sources,guides,resources},website/docs}/**/*`
```

**Review by bflad — CHANGES_REQUESTED** [review]
Overall I think this is okay -- I do believe the glob pattern needs to be updated though. 👍

File: check/directory.go (hunk 1 above)
Comment (DanielMSchmidt) [reply check/directory.go:15]: Oh I see, I wasn't aware :) Thanks for raising this and thank you for checking if it causes trouble down the road :)

File: check/directory.go (hunk 2 above)
Comment (DanielMSchmidt) [reply check/directory.go]: Yeah good point, I slightly tweaked it so that previously ignored files stay ignored.

**Review by bflad — APPROVED** [review]
These changes look good, but there's a new test failure:

```
--- FAIL: TestCheck (0.00s)
    --- FAIL: TestCheck/valid_legacy_directories_requiring_side_navigation (0.00s)
        check_test.go:184: expected error, got no error
```

The ERB side navigation check is very outdated and unnecessary to maintain in this linter at this point so as a followup to this I'll remove all that functionality.

Thanks so much, @DanielMSchmidt 🚀

**Pull request merged**